awriter/signer.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package awriter
+
+import (
+	"archive/tar"
+	"io"
+
+	"github.com/mendersoftware/mender-artifact/artifact"
+	"github.com/pkg/errors"
+)
+
+var ErrAlreadyExistingSignature = errors.New("Will not overwrite existing signature")
+
+// Special fast-track to just sign, nothing else. This skips all the expensive
+// and complicated repacking, and simply adds the manifest.sig file.
+func SignExisting(src io.Reader, dst io.Writer, key []byte, overwrite bool) error {
+	var foundManifest bool
+	rTar := tar.NewReader(src)
+	wTar := tar.NewWriter(dst)
+	buf := make([]byte, 1048576)
```

Comment:
No particular reason, 1MiB just seemed like a good buffer size. Small enough not to blow up memory, and big enough to avoid thrashing on devices with big erase block sizes.

Reply:
I see :)
Suggestion: Either a comment, or something like 2<<19 mby?

Reply:
or 1 << 20 :)

Reply:
I added a comment that it is 1MiB.